src/libMediator/Mediator.cpp

```diff
@@ -157,6 +157,12 @@ bool Mediator::GetIsVacuousEpoch(const uint64_t& epochNum) {
   return ((epochNum + NUM_VACUOUS_EPOCHS) % NUM_FINAL_BLOCK_PER_POW) == 0;
 }
 
+uint32_t Mediator::GetShardNum() const {
```

Comment:
Probably better to move this function to DS?  A bit weird also for `Mediator` to access a mutex owned by `DS`

Reply:
Also, between `shardSize`, `shardNum`, and `numShards`, I think the last one is more easily understood